src/System.Device.Gpio/Interop/Unix/libgpiod/Interop.gpiod_chip_open.cs

```diff
@@ -0,0 +1,13 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Device.Gpio;
+using System.Runtime.InteropServices;
+
+internal partial class Interop
+{
+    [DllImport("libgpiod", SetLastError = true)]
+    internal static extern SafeChipHandle gpiod_chip_open([MarshalAs(UnmanagedType.LPStr)] string path);
```

Comment:
>hip_open([MarshalAs(UnmanagedType.LPStr)] string path [](start = 49, length = 53)

we talked offline about this, I learned on Linux/Mac we always convert the string parameter to UTF8. so I think you don't need [MarshalAs(UnmanagedType.LPStr)] here and you should be fine with what you have in other interop methods which takes the consumer string

Reply:
great thank you for digging into this